### ISSUE TITLE
DSC Pull server problems after May 2017 Patch installation

### ISSUE THREAD

**Issue by skest3qc** [PowerShell/xPSDesiredStateConfiguration#351]
Hi,
after installing May 2017 patches on client like Windows 2008 R2 2012 R2 2016 and updating my pull server with May 2017 patches my pull server stopped working.

I saw this item in the release notes of May 2017 patches:
Deprecated SHA-1 Microsoft Edge and Internet Explorer 11 for SSL/TLS Server Authentication. See Advisory 4010323 for more information.

I am using a self signed certificate (SHA256) and if I run Update-DSCConfiguration I get the following message in eventvwr:
`Job {2093529D-3573-11E7-9113-0050569E2DE2} : 
Http Client cf3d1a0e-c201-4c13-bfc7-98b44a4dc78b failed for WebReportManager for configuration 
FullyQualifiedErrorId :ReportManagerSendStatusReportFailed
 CategoryInfo:InvalidResult: (:) [], InvalidOperationException
 ExceptionMessage:Failed to send status report to the server https://pullserverfqdn:8080/PSDSCPullServer.svc/Node(ConfigurationId='cf3d1a0e-c201-4c13-bfc7-98b44a4dc78b')/SendStatusReport.
, InnerExceptionSystem.AggregateException: One or more errors occurred. ---> System.Net.Http.HttpRequestException: An error occurred while sending the request. ---> System.Net.WebException: The underlying connection was closed: Could not establish trust relationship for the SSL/TLS secure channel. ---> System.Security.Authentication.AuthenticationException: The remote certificate is invalid according to the validation procedure.
   at System.Net.TlsStream.EndWrite(IAsyncResult asyncResult)
   at System.Net.ConnectStream.WriteHeadersCallback(IAsyncResult ar)
   --- End of inner exception stack trace ---
   at System.Net.HttpWebRequest.EndGetRequestStream(IAsyncResult asyncResult, TransportContext& context)
   at System.Net.Http.HttpClientHandler.GetRequestStreamCallback(IAsyncResult ar)
   --- End of inner exception stack trace ---
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at Microsoft.PowerShell.DesiredStateConfiguration.Commands.SendDscStatusCommand.IssueRequest(HttpClient client, String subLink, ErrorRecord& errorRecord)
---> (Inner Exception #0) System.Net.Http.HttpRequestException: An error occurred while sending the request. ---> System.Net.WebException: The underlying connection was closed: Could not establish trust relationship for the SSL/TLS secure channel. ---> System.Security.Authentication.AuthenticationException: The remote certificate is invalid according to the validation procedure.
   at System.Net.TlsStream.EndWrite(IAsyncResult asyncResult)
   at System.Net.ConnectStream.WriteHeadersCallback(IAsyncResult ar)
   --- End of inner exception stack trace ---
   at System.Net.HttpWebRequest.EndGetRequestStream(IAsyncResult asyncResult, TransportContext& context)
   at System.Net.Http.HttpClientHandler.GetRequestStreamCallback(IAsyncResult ar)
   --- End of inner exception stack trace ---<---
`

Does anyone have the same issue?

**Comment by skest3qc** [comment]
It seems that after removing the pullserver and recreating a new self signed certificate after the patch installation it seems to work again...

**Comment by playan** [comment]
Just ran into the same thing. So you uninstalled the Pull servers, and put a new cert on?  I was using a cert issued by our internal CA, guess I can just request a new one. Any ideas why the may patches caused this?

**Comment by mjksgea** [comment]
I ran into this problem where I had pull server with two clients.  The pull server created self signed cert. It worked with one client but not the other.  All machines are 2016 but they were build at different times.  I could not figue it out unitl I saw it in the logs.  I tried different self sighned cert but nothing.  In the end I got Dev domain cert which worked.  The other thing I would like to point out is that I made a mistake with the certificate detials, so I removed it from the VM store and replaced it with a new one.  But the configuration of the Pull server did not remove it even after introducing the new Thumbprint so just be aware.
The error message don't give you much so much time wasted.
mjksgea
